qbraid/runtime/rigetti/job.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (C) 2024 qBraid
+#
+# This file is part of the qBraid-SDK
+#
+# The qBraid-SDK is free software released under the GNU General Public License v3
+# or later. You can redistribute and/or modify it under the terms of the GPL v3.
+# See the LICENSE file in the project root or <https://www.gnu.org/licenses/gpl-3.0.html>.
+#
+# THERE IS NO WARRANTY for the qBraid-SDK, as per Section 15 of the GPL v3.
+
+"""
+Module defining Rigetti job class
+
+"""
+
+import warnings
+from typing import Any, TypeVar
+
+import pyquil.api
+
+from qbraid.runtime.enums import JobStatus
+from qbraid.runtime.exceptions import QbraidRuntimeError
+from qbraid.runtime.job import QuantumJob
+from qbraid.runtime.result import Result
+from qbraid.runtime.result_data import GateModelResultData
+
+T = TypeVar("T")
+"""A generic parameter describing the opaque job handle returned from QAM#execute and subclasses."""
+
+
+class RigettiJobError(QbraidRuntimeError):
+    """Class for errors raised while processing an Rigetti job."""
+
+
+class RigettiJob(QuantumJob):
+    """Rigetti job class."""
+
+    def __init__(
+        self,
+        job_id: str | int,
+        qam: pyquil.api.QAM,
+        execute_response: T,
+        device_id: str,
```

Comment:
The `QuantumJob` already takes an `Optional[qbraid.runtime.QuantumDevice]` this `device_id` param is redundant